magento/pwa-studio
# Layered Navigation - Desktop Optimizations

## Description

- Display the layered navigation contents directly on the category and search pages when on desktop (equal to or above 1024px).
- Changed the markup/layout of the category and search pages to support the left column on desktop.
- Moved and styled the Sort button on desktop. It now also includes the current sorting within the button text.
- Added _X_ Results messaging above the results on category pages (both mobile and desktop)
- Added Show More / Show Less button within the filter list when there are more than 5 items (both mobile and desktop). You can also pass in a new prop "showItems" to the FilterList change this threshold
- Open the first 3 filter accordions on desktop. You can also pass in a new prop "filtersOpen" to the FilterSidebar change this threshold
- Also open any filters that have selected items (both mobile and desktop)
- Scroll to the top of filters (-150px for menu threshold) when changing the filters on desktop

## Related Issue
Closes #3115

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps
1. Go to the Bottoms category page on a desktop browser
2. Verify the layered nav is visible on the left
3. Verify the sort button is on the right and displays current sort order within button
4. Verify the first 3 filters are open and the rest are closed
5. Verify the Show More/Less works correctly within the Color attribute list
6. Verify selecting a filter in one of the closed attributes (ex Size) then forces the accordion open (but can still be manually closed)
7. Verify selecting a filter lower down on the page scrolls the viewport back to the top of the filter list
8. Verify that the mobile filters reflect what was selected on desktop and function as they did before these modifications
9. Search for "pants"
10. Verify the layout is functioning as it did in the category page

## Screenshots / Screen Captures (if appropriate)
![image](https://user-images.githubusercontent.com/68294336/115595609-1f197e00-a2a5-11eb-918a-16095309b870.png)
![image](https://user-images.githubusercontent.com/68294336/115598392-60f7f380-a2a8-11eb-9782-385160f0f80a.png)
![image](https://user-images.githubusercontent.com/68294336/115598440-6fdea600-a2a8-11eb-8463-81c61ae76f5a.png)


## Checklist

[ ] I have added tests to cover my changes, if necessary.
[x] I have updated existing tests to cover my changes, if necessary.
[x] I have added translations for new strings, if necessary.
[x] I have updated the documentation accordingly, if necessary.
